### ISSUE TITLE
P3M GPU fails with MPI

### ISSUE THREAD

**Issue by KonradBreitsprecher** [espressomd/espresso#1451]
`mpirun -np 2 ./pypresso gpup3m_issue.py`

goes down with

```
terminate called after throwing an instance of 'std::out_of_range'
  what():  _Map_base::at
```
Features: 'CUDA', 'ELECTROSTATICS', 'FFTW', 'H5MD', 'P3M'

Testscript:

```
import espressomd
from espressomd.electrostatics import P3M_GPU

S=espressomd.System()
S.time_step=0.01
S.cell_system.skin = 0.4

S.box_l=[10,10,10]

S.part.add(pos=[0,0,0], q=1)
S.part.add(pos=[0,0,2], q=-1)

p3m=P3M_GPU(bjerrum_length=1.0,accuracy=1e-4)
S.actors.add(p3m)
```

**Comment by fweik** [comment]
Funny enough
```
mpirun -np 2 ./pypresso ../testsuite/coulomb_cloud_wall.py
```
works. Will investigate.

**Comment by KonradBreitsprecher** [comment]
On the CD on 19/10, I will work on that. 
#600, #658, #1361 and #239  are related.

**Comment by KonradBreitsprecher** [comment]
Closed via #1546
